### PR TITLE
feat(query): add function-v2numeric sign/trigonometric/abs

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

- Add `sign`, `trigonometric`, `abs` in function-v2
- Add unary `plus`, `minus` op
- Refactor `can_auto_cast` logic which adds: all integer types can cast to int64 and all numeric types can cast to float64

Part of #7255

### CODE REVIEW

File: src/query/functions-v2/src/scalars/math.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        FunctionProperty::default(),
+        || Some(NumberDomain::<f64> { min: PI, max: PI }),
+        |_| Ok(Value::Scalar(PI)),
+    );
+
+    registry.register_1_arg::<NumberType<f64>, NumberType<i8>, _, _>(
+        "sign",
+        FunctionProperty::default(),
+        |_| Some(NumberDomain::<i8> { min: -1, max: 1 }),
+        |val| match val.partial_cmp(&f64::default()) {
```
Comment (andylokandy) [inline src/query/functions-v2/src/scalars/math.rs:64]:
```suggestion
        |val| match val.partial_cmp(&0_f64) {
```

File: src/query/functions-v2/src/scalars/math.rs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        |domain| Some(domain.clone()),
+        |val| val,
+    );
+
+    registry.register_1_arg::<NumberType<i64>, NumberType<u64>, _, _>(
+        "abs",
+        FunctionProperty::default(),
+        |domain| {
+            let max = domain.min.abs().max(domain.max.abs()) as u64;
+            let mut min = domain.min.abs().min(domain.max.abs()) as u64;
```
Comment (andylokandy) [inline src/query/functions-v2/src/scalars/math.rs:83]: abs could panic, I suggest using unsigned_abs instead

File: src/query/functions-v2/tests/it/scalars/parser.rs [hunk 3]
```diff
@@ -212,7 +212,8 @@ pub fn transform_literal(lit: ASTLiteral) -> Literal {
         ASTLiteral::String(s) => Literal::String(s.as_bytes().to_vec()),
         ASTLiteral::Boolean(b) => Literal::Boolean(b),
         ASTLiteral::Null => Literal::Null,
-        _ => unimplemented!(),
+        ASTLiteral::Float(f) => Literal::Float64(f),
+        _ => unimplemented!("{}", lit),
```
Comment (andylokandy) [inline src/query/functions-v2/tests/it/scalars/parser.rs:216]:
```suggestion
        _ => unimplemented!("{lit}"),
```

File: src/query/functions-v2/src/scalars/math.rs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        "pi",
+        FunctionProperty::default(),
+        || Some(NumberDomain::<f64> { min: PI, max: PI }),
+        |_| Ok(Value::Scalar(PI)),
+    );
+
+    registry.register_1_arg::<NumberType<f64>, NumberType<i8>, _, _>(
+        "sign",
+        FunctionProperty::default(),
+        |_| Some(NumberDomain::<i8> { min: -1, max: 1 }),
```
Comment (andylokandy) [inline src/query/functions-v2/src/scalars/math.rs:63]: maybe can be more precise.

**Comment by drmingdrmer** [comment]
`main` branch is force pushed(**again** `:(`, to cleanup big object in history), update your working branch please: [To everybody: main branch is force pushed. #7263](https://github.com/datafuselabs/databend/issues/7263)

File: src/query/functions-v2/src/scalars/arithmetic.rs [hunk 5]
```diff
@@ -26,7 +26,7 @@ use super::arithmetic_modulo::vectorize_modulo;
 
 pub fn register(registry: &mut FunctionRegistry) {
     registry.register_aliases("plus", &["add"]);
-    registry.register_aliases("minus", &["substract"]);
+    registry.register_aliases("minus", &["substract", "neg"]);
     registry.register_aliases("div", &["intdiv"]);
 
     // TODO support modulo
```
Comment (andylokandy) [inline src/query/functions-v2/src/scalars/arithmetic.rs:32]: Is this outdated

File: src/query/functions-v2/src/scalars/math.rs [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+        FunctionProperty::default(),
+        |_| None,
+        |f: f64| f.tan(),
+    );
+
+    registry.register_1_arg::<NumberType<f64>, NumberType<f64>, _, _>(
+        "cot",
+        FunctionProperty::default(),
+        |_| None,
+        |f: f64| 1.0f64 / f.tan(),
```
Comment (andylokandy) [inline src/query/functions-v2/src/scalars/math.rs:59]: will this panic?

**Review by andylokandy — APPROVED** [review]

**Review by PsiACE — APPROVED** [review]

**Pull request merged**